### PR TITLE
added sample integration between graphql-jit and apollo

### PR DESCRIPTION
**Pull request by wojtkiewicz**
Related to: https://github.com/zalando-incubator/graphql-jit/issues/67

This PR contains sample integration between graphql-jit and Apollo Server. It can be used as a starting point to build more robust `apollo-server-graphql-jit` component in the future. What you probably want is to add metrics and logs for instance.

For legal reasons I can't contribute entire module that I was running on production (at least not yet, it has to pass lengthy security clearance to be open sourced).

### CODE REVIEW

File: examples/blog-apollo-server/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "blog-apollo-server",
+  "version": "0.1.0",
+  "dependencies": {
+    "apollo-server": "^2.9.16",
+    "graphql-jit": "^0.4.2",
```
Comment (boopathi) [inline examples/blog-apollo-server/package.json]: point to the GraphQL-jit from `../../` (or) remove this dependency and replace imports to use `../../`

File: examples/blog-apollo-server/src/executor.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  cacheSize = 1024,
+  compilerOpts: Partial<CompilerOptions> = {}
+): GraphQLExecutor => {
+  const cache = LRU<CompiledQuery>(cacheSize);
+  return async ({ context, document, operationName, request, queryHash }) => {
+    const prefix = operationName || "NotParametrized";
+    const cacheKey = `${prefix}-${queryHash}`;
+    let compiledQuery = cache.get(cacheKey);
+    if (!compiledQuery) {
+      const compilationResult = compileQuery(schema, document, operationName || undefined, compilerOpts);
```
Comment (boopathi) [inline examples/blog-apollo-server/src/executor.ts]:
```suggestion
      const compilationResult = compileQuery(schema, document, operationName, compilerOpts);
```

Is there a valid case where ` || undefined` is useful?

File: examples/blog-apollo-server/package.json (hunk 1 above)
Comment (wojtkiewicz) [reply examples/blog-apollo-server/package.json]: Done. I had to upgrade `graphql` to `^14.5.3` for this code to compile though. Is this ok with you?

File: examples/blog-apollo-server/src/executor.ts (hunk 2 above)
Comment (wojtkiewicz) [reply examples/blog-apollo-server/src/executor.ts]: Unfortunately apollo defines operationName as `string | null` and graphql-jit requires `string | undefined`. So I need this extra `null => undefined` conversion to compile.

File: examples/blog-apollo-server/package.json (hunk 1 above)
Comment (ruiaraujo) [reply examples/blog-apollo-server/package.json]: The upgrade is okay, but I wonder why it was required?

File: examples/blog-apollo-server/src/executor.ts (hunk 2 above)
Comment (ruiaraujo) [reply examples/blog-apollo-server/src/executor.ts]: @boopathi we should extend what we expect. We dont care if it is null or undefined.

File: examples/blog-apollo-server/package.json (hunk 1 above)
Comment (wojtkiewicz) [reply examples/blog-apollo-server/package.json]: Main culprit in this case is `GraphQLSchema`. Graphql-jit was expecting older model and apparently there are some differences that compiler detected. Downgrading `apollo-server` should also help if you need more testing with latest `graphql-js`.

**Comment by ruiaraujo** [comment]
:+1:

**Comment by boopathi** [comment]
👍

**Pull request merged**